sdk/tables/azure-tables/review/tables.api.md

```diff
@@ -4,8 +4,595 @@
 
 ```ts
 
+import * as coreHttp from '@azure/core-http';
+
+// @public
+export interface AccessPolicy {
+    expiry: Date;
+    permission: string;
+    start: Date;
+}
+
+// @public
+export interface CorsRule {
+    allowedHeaders: string;
+    allowedMethods: string;
+    allowedOrigins: string;
+    exposedHeaders: string;
+    maxAgeInSeconds: number;
+}
+
+// @public
+export interface GeneratedClientOptionalParams extends coreHttp.ServiceClientOptions {
+    endpoint?: string;
+    version?: string;
+}
+
 // @public (undocumented)
-export function helloWorld(): string;
+export interface GeoReplication {
+    lastSyncTime: Date;
+    status: GeoReplicationStatusType;
+}
+
+// @public
+export type GeoReplicationStatusType = "live" | "bootstrap" | "unavailable";
+
+// @public
+export interface Logging {
+    delete: boolean;
+    read: boolean;
+    retentionPolicy: RetentionPolicy;
+    version: string;
+    write: boolean;
+}
+
+// @public (undocumented)
+export interface Metrics {
+    enabled: boolean;
+    includeAPIs?: boolean;
+    retentionPolicy?: RetentionPolicy;
+    version?: string;
+}
+
+// @public
+export type OdataMetadataFormat = "application/json;odata=nometadata" | "application/json;odata=minimalmetadata" | "application/json;odata=fullmetadata";
+
+// @public
+export interface QueryOptions {
+    filter?: string;
+    format?: OdataMetadataFormat;
+    select?: string;
+    top?: number;
+}
+
+// @public
+export type ResponseFormat = "return-no-content" | "return-content";
+
+// @public
+export interface RetentionPolicy {
+    days?: number;
+    enabled: boolean;
+}
+
+// @public
+export interface ServiceGetPropertiesHeaders {
+    // (undocumented)
+    xMsClientRequestId?: string;
+    // (undocumented)
+    xMsRequestId?: string;
+    // (undocumented)
+    xMsVersion?: string;
+}
+
+// @public
+export interface ServiceGetPropertiesOptionalParams extends coreHttp.OperationOptions {
+    requestId?: string;
+    timeout?: number;
+}
+
+// @public
+export type ServiceGetPropertiesResponse = ServiceGetPropertiesHeaders & TableServiceProperties & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: TableServiceProperties;
+        parsedHeaders: ServiceGetPropertiesHeaders;
+    };
+};
+
+// @public
+export interface ServiceGetStatisticsHeaders {
+    // (undocumented)
+    date?: Date;
+    // (undocumented)
+    xMsClientRequestId?: string;
```

Comment:
In other storage libraries, these `xMs*` headers all have their `x-ms-client-name`s. For example,
https://github.com/Azure/azure-rest-api-specs/blob/2adcc6851f4854ca17227f976a65ebbfba76ab42/specification/storage/data-plane/Microsoft.BlobStorage/preview/2019-12-12/blob.json#L65

Can we fix the swagger?

The generated code then looks like https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-blob/src/generated/src/models/index.ts#L3374

```typescript
export interface ServiceGetStatisticsHeaders {
  /**
   * If a client request id header is sent in the request, this header will be present in the
   * response with the same value.
   */
  clientRequestId?: string;
...
```

Reply:
Not blocking PR but something we should consider before releasing.

Reply:
Good catch! The Swagger does define `x-ms-client-name`, after taking a look this seems to be caused by the same Autorest issue which is dropping the descriptions, Azure/autorest.modelerfour/pull/318 should address this as well